src/resolvers/default.rs

```diff
@@ -128,21 +128,11 @@ impl Random for OsRng {}
 
 impl Dh25519 {
     fn derive_pubkey(&mut self) {
-        // TODO: use `MontgomeryPoint::mul_base` in final v4 release of curve25519-dalek
-        // See dalek-cryptography/curve25519-dalek#503
-        let point = EdwardsPoint::mul_base(&self.privkey).to_montgomery();
+        let point = X25519_BASEPOINT.mul_clamped(self.privkey);
```

Comment:
It would be faster to use `mul_base_clamped` which can use precomputed basepoint tables:

```suggestion
        let point = MontgomeryPoint::mul_base_clamped(self.privkey);
```

Reply:
Thanks! Will add a new commit with this change (would merge yours except I also have to remove the import).